opensafely/__init__.py

```diff
@@ -81,27 +78,25 @@ def add_subcommand(cmd, module):
         except Exception:
             pass
 
-    # start by using looser parsing only known args, so we can get the sub command
-    args, unknown = parser.parse_known_args()
-
-    if args.handles_unknown_args:
-        kwargs = vars(args)
-        kwargs["unknown_args"] = unknown
-    # allow supparsers to opt out of looser parser_known_args parsing
-    else:
-        # reparse args with stricter semantics
-        args = parser.parse_args()
-        kwargs = vars(args)
+    args = parser.parse_args()
+    kwargs = vars(args)
 
     function = kwargs.pop("function")
-    kwargs.pop("handles_unknown_args")
     success = function(**kwargs)
 
     # if `run`ning locally, run `check` in warn mode
     if function == local_run.main and "format-output-for-github" not in kwargs:
         check.main(continue_on_error=True)
 
-    sys.exit(0 if success is not False else 1)
+    # allow functions to return True/False, or an explicit exit code
+    if success is False:
+        exit_code = 1
+    elif success is True:
```

Comment:
I think this needs `or success is None` to fully replicate the existing behaviour where functions which don't return anything are assumed successful.